torch_geometric/nn/conv/pna_conv.py

```diff
@@ -196,7 +196,8 @@ def __repr__(self):
     def get_degree_histogram(loader: DataLoader) -> Tensor:
         r"""Returns the degree histogram to be used as input for the :obj:`deg`
         argument in :class:`PNAConv`."""
-        deg_histogram = torch.zeros(1, dtype=torch.long)
+        device = next(iter(loader)).edge_index.device
```

Comment:
Can we obtain the device without iterating through the dataloader? This seems to introduce additional costs.

Reply:
I tried different approaches, but this seems to be the most reliable, as we often lack some members in the loader.
As for performance, there is some cost, but when compared to the whole `get_degree_histogram` function it is a really small fraction. As an alternative, I can propose passing `device` as an argument. WDYT?

Reply:
Looks pretty good to me. Thank you @DamianSzwichtenberg and @rusty1s